src/main/java/org/eln2/mc/common/content/VoltageSource.kt

```diff
@@ -0,0 +1,107 @@
+package org.eln2.mc.common.content
+
+import mcp.mobius.waila.api.IPluginConfig
+import net.minecraft.resources.ResourceLocation
+import net.minecraft.world.phys.Vec3
+import org.ageseries.libage.sim.electrical.mna.Circuit
+import org.ageseries.libage.sim.electrical.mna.component.VoltageSource
+import org.eln2.mc.Eln2
+import org.eln2.mc.client.render.PartialModels
+import org.eln2.mc.client.render.foundation.BasicPartRenderer
+import org.eln2.mc.common.cells.CellRegistry
+import org.eln2.mc.common.cells.foundation.CellBase
+import org.eln2.mc.common.cells.foundation.CellPos
+import org.eln2.mc.common.cells.foundation.Conventions
+import org.eln2.mc.common.cells.foundation.objects.*
+import org.eln2.mc.common.parts.foundation.CellPart
+import org.eln2.mc.common.parts.foundation.ConnectionMode
+import org.eln2.mc.common.parts.foundation.IPartRenderer
+import org.eln2.mc.common.parts.foundation.PartPlacementContext
+import org.eln2.mc.common.space.RelativeRotationDirection
+import org.eln2.mc.extensions.TooltipBuilderExtensions.voltageSource
+import org.eln2.mc.integration.waila.IWailaProvider
+import org.eln2.mc.integration.waila.TooltipBuilder
+
+/**
+ * The voltage source object has a bundle of resistors, whose External Pins are exported to other objects, and
+ * a voltage source, connected to the Internal Pins of the bundle.
+ * */
+class VoltageSourceObject : ElectricalObject(), IWailaProvider {
+    private lateinit var source: VoltageSource
+    val hasSource get() = this::source.isInitialized
+
+    private val resistors = ResistorBundle(0.01)
+
+    /**
+     * Gets or sets the potential of the voltage source.
+     * */
+    var potential: Double = 100.0
```

Comment:
Nit: If you're going to start the resistors off at one ohm, probably a good idea to start these at one volt.